### PR TITLE
Consolidate node datastores: part 2

### PR DESCRIPTION
**Pull request by md2119**
## Description

This PR replaces global node datastore in search. 

Part 1: #4309 

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### CODE REVIEW

File: central/node/globaldatastore/datastore.go [hunk 1]
```diff
@@ -10,6 +10,7 @@ import (
 )
 
 // GlobalDataStore is the global datastore for all nodes across all clusters.
+//
```
Comment (RTann) [inline central/node/globaldatastore/datastore.go:13]: guessing you're using go1.19 locally?

File: central/graphql/resolvers/resolver.go [hunk 2]
```diff
@@ -142,7 +144,8 @@ func New() *Resolver {
 		NamespaceDataStore:          namespaceDataStore.Singleton(),
 		NetworkPoliciesStore:        npDS.Singleton(),
 		NetworkFlowDataStore:        nfDS.Singleton(),
-		NodeGlobalDataStore:         nodeDataStore.Singleton(),
+		NodeGlobalDataStore:         nodeGlobalDataStore.Singleton(),
```
Comment (RTann) [inline central/graphql/resolvers/resolver.go:147]: is this needed anymore?

File: central/node/globaldatastore/datastore.go (hunk 1 above)
Comment (md2119) [reply central/node/globaldatastore/datastore.go:13]: :)

File: central/graphql/resolvers/resolver.go (hunk 2 above)
Comment (md2119) [reply central/graphql/resolvers/resolver.go:147]: Still some instances, which will be addressed in follow ups. This PR focuses on global search functionality.

**Review by theencee — APPROVED** [review]

**Pull request merged**